config/opal_check_package.m4

```diff
@@ -50,6 +50,15 @@ AC_DEFUN([_OPAL_CHECK_PACKAGE_HEADER], [
                   [# no go on the as is - reset the cache and try again
                    unset opal_Header])])
 
+    AS_IF([test "$opal_check_package_header_happy" = "no"],
+          [AS_IF([test "$dir_prefix" != ""],
+                 [$1_CPPFLAGS="$$1_CPPFLAGS -I$dir_prefix"
+                  CPPFLAGS="$CPPFLAGS -I$dir_prefix"])
+          AC_CHECK_HEADERS([$2], [opal_check_package_header_happy="yes"], [], [$6])
```

Comment:
@pkovacs @jsquyres @rhc54 @ggouaillardet @yosefe @jladd-mlnx This check is wrong when the package headers are installed in /usr/include  but configure was called with the specific package dir. The check will pickup h file from the /usr/include and add -I$dir_prefix to the CPPFLAGS which is wrong.

Reply:
OK, so we can add `/usr/include` and `/usr/local/include` filters to the to code block above that (lines 44, 45).  That will handle the `--with-foo=/usr/include` and ``--with-foo='/usr/include/local` cases without adding unecessary CPPFLAGS
,
```
    AS_IF([test "$dir_prefix" = "/usr" || \
           test "$dir_prefix" = "/usr/local || \
           test "$dir_prefix" = "/usr/include || \
           test "$dir_prefix" = "/usr/local/include"],
```

Reply:
Added https://github.com/open-mpi/ompi/pull/4722 to correct the concern above.